components/core/src/clp_s/JsonParser.cpp

```diff
@@ -557,6 +621,307 @@ int32_t JsonParser::add_metadata_field(std::string_view const field_name, NodeTy
+            }
+            default:
+                // Don't need to add value for obj or null
+                break;
+        }
+        m_current_schema.insert_ordered(node_id);
+    }
+
+    int32_t const current_schema_id = m_archive_writer->add_schema(m_current_schema);
+    m_current_parsed_message.set_id(current_schema_id);
+    m_archive_writer->append_message(current_schema_id, m_current_schema, m_current_parsed_message);
+}
+
+auto JsonParser::parse_from_ir() -> bool {
+    for (auto& file_path : m_file_paths) {
+        auto const fsize = std::filesystem::file_size(file_path);
+        if (0 == fsize) {
+            m_archive_writer->close();
+            return false;
+        }
+        clp::streaming_compression::zstd::Decompressor decompressor;
+        size_t curr_pos = 0;
+        size_t last_pos = 0;
+        decompressor.open(file_path);
+
+        auto deserializer_result{Deserializer<IrUnitHandler>::create(decompressor, IrUnitHandler{})
+        };
+        if (deserializer_result.has_error()) {
+            decompressor.close();
+            m_archive_writer->close();
+            return false;
+        }
+        auto& deserializer = deserializer_result.value();
+        auto& ir_unit_handler{deserializer.get_ir_unit_handler()};
+
+        int32_t log_event_idx_node_id{};
+        auto add_log_event_idx_node = [&]() {
+            if (m_record_log_order) {
+                log_event_idx_node_id
+                        = add_metadata_field(constants::cLogEventIdxName, NodeType::Integer);
+            }
+        };
+        add_log_event_idx_node();
+        while (true) {
+            auto const kv_log_event_result{deserializer.deserialize_next_ir_unit(decompressor)};
+
+            if (kv_log_event_result.has_error()) {
+                break;
+            }
+            if (kv_log_event_result.value() == clp::ffi::ir_stream::IrUnitType::EndOfStream) {
+                break;
+            }
+            if (kv_log_event_result.value() == clp::ffi::ir_stream::IrUnitType::LogEvent) {
+                auto const kv_log_event = &(ir_unit_handler.get_deserialized_log_event().value());
+
+                m_current_schema.clear();
+
+                // Add log_event_idx field to metadata for record
+                if (m_record_log_order) {
+                    m_current_parsed_message.add_value(
+                            log_event_idx_node_id,
+                            m_archive_writer->get_next_log_event_id()
+                    );
+                    m_current_schema.insert_ordered(log_event_idx_node_id);
+                }
+
+                try {
+                    parse_kv_log_event(*kv_log_event);
+                } catch (std::exception const& e) {
+                    SPDLOG_ERROR("Encountered error while parsing a kv log event - {}", e.what());
+                    decompressor.try_get_pos(curr_pos);
+                    m_archive_writer->increment_uncompressed_size(curr_pos - last_pos);
```

Comment:
```suggestion
m_archive_writer->close();
```
No need to update metadata on the failure path. Also missed it earlier but we should close the archive on the error path to be consistent with JSON ingestion behaviour.